crates/utils/src/utils/markdown.rs

```diff
@@ -1,3 +1,131 @@
+        "> #### Hello\n > \n > - Hola\n > - 안영\n",
+        "<blockquote>\n<h4>Hello</h4>\n<ul>\n<li>Hola</li>\n<li>안영</li>\n</ul>\n</blockquote>\n"
+      ),
+      (
+        "ordered lists",
+        "1. pen\n2. apple\n3. apple pen",
+        "<ol>\n<li>pen</li>\n<li>apple</li>\n<li>apple pen</li>\n</ol>\n"
+      ),
+      (
+        "unordered lists",
+        "- pen\n- apple\n- apple pen",
+        "<ul>\n<li>pen</li>\n<li>apple</li>\n<li>apple pen</li>\n</ul>\n"
+      ),
+      (
+        "code",
+        "this is my amazing `code snippet`",
+        "<p>this is my amazing <code>code snippet</code></p>\n"
+      ),
+      (
+        "code block",
+        "this is my amazing ```code block```",
+        "<p>this is my amazing <code>code block</code></p>\n"
+      ),
+      (
+        "links",
+        "[Lemmy](https://join-lemmy.org/ \"Join Lemmy!\")",
+        "<p><a href=\"https://join-lemmy.org/\" title=\"Join Lemmy!\">Lemmy</a></p>\n"
+      ),
+      (
+        "images",
+        "![My linked image](https://image.com \"image alt text\")",
+        "<p><img src=\"https://image.com\" alt=\"My linked image\" title=\"image alt text\"></p>\n"
+      ),
+      (
+        "invalid spoiler",
+        "::: spoiler click to see more\nbut I never finished",
+        "<p>::: spoiler click to see more\nbut I never finished</p>\n",
+      ),
+      (
+        "another invalid spoiler",
+        "::: spoiler\nnever added the lead in\n:::",
+        "<p>::: spoiler\nnever added the lead in\n:::</p>\n",
+      ),
+      (
+        "basic spoiler, but no newline at the end",
+        "::: spoiler click to see more\nhow spicy!\n:::",
+        "<details><summary>click to see more</summary><p>how spicy!\n</p></details>\n"
+      ),
+      (
+        "basic spoiler with a newline at the end.",
+        "::: spoiler click to see more\nhow spicy!\n:::\n",
+        "<details><summary>click to see more</summary><p>how spicy!\n</p></details>\n"
+      ),
+      (
+        "spoiler with extra markdown on the call to action. No special parsing will be done.",
+        "::: spoiler _click to see more_\nhow spicy!\n:::\n",
+        "<details><summary>_click to see more_</summary><p>how spicy!\n</p></details>\n"
+      ),
+      (
+        "spoiler with extra markdown in the fenced spoiler block.",
+        "::: spoiler click to see more\n**how spicy!**\n*i have many lines*\n:::\n",
+        "<details><summary>click to see more</summary><p><strong>how spicy!</strong>\n<em>i have many lines</em>\n</p></details>\n"
+      ),
+      (
+        "spoiler mixed with other content.",
+        "hey you\npsst, wanna hear a secret?\n::: spoiler lean in and i'll tell you\n**you are breathtaking!**\n:::\nwhatcha think about that?",
+        "<p>hey you\npsst, wanna hear a secret?</p>\n<details><summary>lean in and i'll tell you</summary><p><strong>you are breathtaking!</strong>\n</p></details>\n<p>whatcha think about that?</p>\n"
+      ),
+    ];
```

Comment:
This seems like a lot of tests, can you shorten it a bit? It should only have to check that markdown works in general and plugins are working, but the main functionality should be tested upstream already.

Reply:
I combined some tests and moved the spoiler-specific ones to be with the plugin. Hopefully this doesn't look as cumbersome as before!